android/app/src/main/kotlin/com/pockeat/MainActivity.kt

```
@@ -13,110 +15,137 @@ import com.pockeat.widget.CustomHomeWidgetPlugin
+            val intent = packageManager.getLaunchIntentForPackage(HEALTH_CONNECT_PACKAGE)
+            if (intent != null) {
+                startActivity(intent)
+                result.success(true)
+            } else {
+                // Try an alternate approach
+                val settingsIntent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+                settingsIntent.data = Uri.parse("package:$HEALTH_CONNECT_PACKAGE")
+                startActivity(settingsIntent)
+                result.success(true)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)